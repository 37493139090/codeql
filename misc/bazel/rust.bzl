load("@rules_rust//rust:defs.bzl", "rust_binary")

def codeql_rust_binary(
        name,
        target_compatible_with = None,
        visibility = None,
        symbols_test = True,
        **kwargs):
    # rust_label_name = name + "_single_arch"
    # universal_binary(
    #     name = name,
    #     dep = ":" + rust_label_name,
    #     target_compatible_with = target_compatible_with,
    #     visibility = visibility,
    # )
    rust_binary(
        name = name,
        target_compatible_with = target_compatible_with,
        visibility = visibility,
        **kwargs
    )
    # if symbols_test:
    #     glibc_symbols_check(name = name + "symbols-test", binary = name)
