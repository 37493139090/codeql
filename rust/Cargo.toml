[workspace]

resolver = "2"
members = [
    "extractor",
    "extractor/macros",
    "ast-generator",
]

[patch.crates-io]
# patch for build script bug preventing bazel build
# see https://github.com/rust-lang/rustc_apfloat/pull/17
rustc_apfloat = { git = "https://github.com/redsun82/rustc_apfloat.git", rev = "096d585100636bc2e9f09d7eefec38c5b334d47b" }
