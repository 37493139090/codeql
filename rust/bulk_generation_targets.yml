strategy: dca
language: rust
destination: rust/ql/lib/ext/generated
targets:
- name: rust
- name: libc
- name: log
- name: memchr
- name: once_cell
- name: rand
- name: smallvec
- name: serde
- name: tokio
- name: reqwest
- name: rocket
- name: actix-web
- name: hyper
- name: clap
