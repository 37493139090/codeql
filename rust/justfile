import '../justfile'

install LANGUAGE='rust': (_install LANGUAGE)
build LANGUAGE='rust': (_build LANGUAGE)
