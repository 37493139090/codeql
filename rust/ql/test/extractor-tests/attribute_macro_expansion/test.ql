import rust
import TestUtils

from Item i, MacroItems items, Item expanded
where toBeTested(i) and i.getExpanded() = items and items.getAnItem() = expanded
select i, expanded
