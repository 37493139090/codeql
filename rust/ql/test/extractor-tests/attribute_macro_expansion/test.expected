| attr_macro_expansion.rs:1:1:2:11 | fn foo | attr_macro_expansion.rs:2:4:2:6 | Static |
| attr_macro_expansion.rs:1:1:2:11 | fn foo | attr_macro_expansion.rs:2:4:2:10 | fn foo |
